### PR TITLE
NAS-128814 / 24.10 / Remove legacy samba domain support

### PR DESCRIPTION
**Pull request by anodos325**
DragonFish was last release where we support this legacy feature. This commit removes most of middleware code, but leaves database untouched. Next related PR for FreeIPA support will finalize schema changes.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/smb_/util_smbconf.py [hunk 1]
```diff
@@ -262,17 +234,6 @@ def generate_smb_conf_dict(
                 else:
                     domain = smb_service_config['workgroup']
             case 'DS_TYPE_LDAP':
-                if not ldap_enabled or not lc['has_samba_schema']:
```
Comment (mgrimesix) [inline src/middlewared/middlewared/plugins/smb_/util_smbconf.py:265]: Maybe add a comment why this is an empty case.
We _could_ add a log message if something is unexpectedly calling it.

**Review by mgrimesix — APPROVED** [review]
99% removal.

File: src/middlewared/middlewared/plugins/ldap.py [hunk 2]
```diff
@@ -263,7 +263,6 @@ class Config:
         Int('dns_timeout', default=5),
         Int('kerberos_realm', null=True),
         Str('kerberos_principal'),
-        Bool('has_samba_schema', default=False),
```
Comment (bmeagherix) [inline src/middlewared/middlewared/plugins/ldap.py:266]: Just a question.  I believe removing this equates to a public API change, and can see the change in docs for the `do_update`.  Is this all that needs to occur for a public API change?

Comment (anodos325) [reply src/middlewared/middlewared/plugins/ldap.py:266]: Yes. That's why one needs to be somewhat careful about changing a public accepts decorator.

**Pull request merged**